test/spot-request-poll_test.js

```diff
@@ -33,131 +32,183 @@ describe('Spot Request Polling', () => {
     sandbox.restore();
   });
 
-  it('no outstanding spot requests', async () => {
-    await pollSpotRequests({ec2: {}, region: defaultSR.region, state, runaws: () => {}});
-  });
+  describe('constructor', () => {
+
+    it('succeeds with valid args', () => {
+      const poller = new SpotRequestPoller({ec2: {}, regions: [defaultSR.region], state, runaws: () => {}});
+    });
+
+    it('fails with non-list region arg', () => {
+      try {
+        const poller = new SpotRequestPoller({ec2: {}, regions: defaultSR.region, state, runaws: () => {}});
+        return Promise.reject(Error('Line should not be reached'));
+      } catch(e) { }
+    });
+
+    it('fails with list containing non-string regions', () => {
+      try {
+        const poller = new SpotRequestPoller({ec2: {}, regions: [defaultSR.region, 1], state, runaws: () => {}});
+        return Promise.reject(Error('Line should not be reached'));
+      } catch(e) { }
+    });
```

Comment:
I think [`assert.throws`](https://nodejs.org/api/assert.html#assert_assert_throws_block_error_message) is the way here.